### PR TITLE
#185218925 Ft Admin should be able to disable an account

### PR DESCRIPTION
**Pull request by jcharles921**
## What does this PR do?
This PR create a dashboard for the administrators to be able to manage accounts .
## Description of Task to be completed?
Admin dashboard should list all users with the option of disabling accounts 
Due to varying reasons e.g fraud, abuse, malicious use or violation of terms of service, admin should be able to disable any user account. 
## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout <branch-name>`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm run dev`

- Check the application's documentation on Storybook: `npm run storybook`

## What are the relevant pivotal tracker/Trello stories?
- [#185218926](https://www.pivotaltracker.com/story/show/185218926)

## Screenshots (if appropriate)
image

image

https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/72489391/c8c5f621-f5d9-4100-80b1-060555ee70c9

### CODE REVIEW

**Review by Dyves1 — APPROVED** [review]

**Review by bergad37 — COMMENTED** [review]
Well done, is ok that you did not use the redux toolkit!?

File: src/App.jsx [hunk 1]
```diff
@@ -30,6 +31,10 @@ const App = () => {
           <Route path='/login' element={<LoginPage />} />
           <Route path='/signup' element={<SingupPage />} />
           <Route path='/product/:id' element={<SingleProductPage />} />
+          <Route
+            path='/dashboard/users'
+            element={<AdminManageUserContainer />}
```
Comment (niyontwali) [inline src/App.jsx]: Please only pages should appear here, rename or create another file. Follow the conventions we agreed on.

File: src/containers/Userlist.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  reset
+} from '../states/features/users/usersSlice';
+import Loading from '../components/Loading';
+
+export const Userlist = ({ userList }) => {
+  const dispatch = useDispatch();
+  const { isLoading, accountStatus, message } = useSelector((state) => {
+    return state.users;
+  });
+  const [open, setOpen] = React.useState(false);
```
Comment (niyontwali) [inline src/containers/Userlist.jsx]: why `React.useState` yet you already imported useState. You only do that if you didn't distructure useState in the import

File: src/containers/Userlist.jsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    const handleClose = (reason) => {
+      if (reason === 'clickaway') {
+        return;
+      }
+      setOpen(false);
+    };
+
+    if (accountStatus) {
+      dispatch(disableUser(e.target.id));
+      console.log(message);
```
Comment (niyontwali) [inline src/containers/Userlist.jsx]: remove also console

File: src/containers/Userlist.jsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                  severity='error'
+                  style={{ fontSize: '20px', padding: '6px 8px' }}
+                >
+                  {message}
+                </Alert>
+              </Snackbar>
+      )}
+      {isLoading && <Loading />}
+      {userList?.map((user) => (
+        /* eslint-disable-next-line */
```
Comment (niyontwali) [inline src/containers/Userlist.jsx]: remove this eslint comment. If throws errors try to see how you can resolve it not disabling the rules.

File: src/containers/Userlist.jsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+              sx={{ color: 'white' }}
+            >
+              <MenuItem value={1}>Admin</MenuItem>
+              <MenuItem value={2}>Buyer</MenuItem>
+              <MenuItem value={3}>Seller</MenuItem>
+            </Select>
+          </span>
+
+          <button className='image-button flex justify-center ' type='button'>
+            {/* eslint-disable-next-line */}
```
Comment (niyontwali) [inline src/containers/Userlist.jsx]: The same applies to this

File: src/containers/UsersContainer.jsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+          )}
+          {networkError && (
+            <div className='loading_div w-full'>
+              Server or Network Error{' '}
+              <FontAwesomeIcon icon='fa-solid fa-exclamation-circle' />
+            </div>
+          )}
+          {{ userList } && <Userlist userList={userList} />}
+          <div className='pagination'>
+            {/* eslint-disable-next-line */}
```
Comment (niyontwali) [inline src/containers/UsersContainer.jsx]: The same applies to this

File: src/containers/Userlist.jsx (hunk 5 above)
Comment (niyontwali) [inline src/containers/Userlist.jsx]: The same applies to this

File: src/containers/UsersContainer.jsx (hunk 6 above)
Comment (niyontwali) [inline src/containers/UsersContainer.jsx]: The same applies to this

File: src/containers/UsersNavigationBigScreen.jsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import DashButton from '../components/DashButton';
+
+const NavigationDashbooard = () => {
+  // const User= "Admin";
```
Comment (niyontwali) [inline src/containers/UsersNavigationBigScreen.jsx]: Remove this line of commented code

File: src/containers/UsersNavigationSmallScreen.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import DashButton from '../components/DashButton';
+
+const NavigationDashboardSmall = () => {
+  // const User= "Admin";
```
Comment (niyontwali) [inline src/containers/UsersNavigationSmallScreen.jsx:7]: This too

File: src/states/features/users/usersSlice.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-use-before-define */
```
Comment (niyontwali) [inline src/states/features/users/usersSlice.js]: Please remove these eslint-disable comments

**Review by niyontwali — CHANGES_REQUESTED** [review]

**Comment by niyontwali** [comment]
One last comment, please have some space(padding-bottom) to your pagnation, it should usually be equal to the padding top of the table. Another thing when data is being fetched (i.e when the loader is active) please make sure the pagination also is not visible, otherwise it makes no sense to see the pagination when no data is yet displayed. Thanks

File: src/App.jsx (hunk 1 above)
Comment (jcharles921) [reply src/App.jsx]: ok

File: src/states/features/users/usersSlice.js (hunk 9 above)
Comment (jcharles921) [reply src/states/features/users/usersSlice.js]: For the padding and pagination
https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/72489391/0921edb1-5ff1-4eb1-addd-6075a8f1fc20

**Review by irakevine — APPROVED** [review]

**Comment by mken1205** [comment]
Great work

**Review by mken1205 — APPROVED** [review]

File: vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs
Comment (nishimweprince) [inline vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs:1]: Consider removing this file or ignore it not to be tracked by git.

**Review by nishimweprince — CHANGES_REQUESTED** [review]
I added a comment

**Review by niyontwali — APPROVED** [review]

**Review by nishimweprince — APPROVED** [review]

**Pull request merged**